HFSSdrawpy/core/port.py

```diff
@@ -183,3 +184,49 @@ def rotate_ports(ports, angle):
             posx = port.pos[0]*np.cos(rad)+port.pos[1]*np.sin(-rad)
             posy = port.pos[0]*np.sin(rad)+port.pos[1]*np.cos(rad)
             port.pos = Vector([posx, posy])
+
+    def split(self, subnames=[], gap=None):
+        """
+        Creates CPW like ports out of the listed subports
+
+        Parameters
+        ----------
+        subnames : list of str, must be part of eponymous instance attribute
+                   if empty, split all ports
+                   if -1 is passed, split all ports but last
+        gap      : str, gap size for the output CPW ports
+                   if None, track size is used
+        Returns
+        -------
+        subports : list of Port instances
+
+        """
+
+        if not subnames:
+            subnames = self.subnames
+        elif subnames == -1:
+            subnames = self.subnames[:-1]
```

Comment:
I guess this assumes the last subport defined is the outline/gap. This assumption is not ideal but I guess this is the best solution for now.